apps/apps.humanatlas.io/src/app/pages/us2/services/tissue-origin.service.ts

```diff
@@ -0,0 +1,188 @@
+  }
+
+  /**
+   * Returns user selection for organ and tool
+   * @returns UserSelection Object
+   */
+  getSelections(): UserSelection {
+    return this.selections;
+  }
+}
+
+/**
+ * Resolver for Tissue Origin Predictions page
+ * @returns Tissue origin predictions result observable
+ */
+export function resolveTissueOriginPredictions(): Observable<TissueOriginPredictions> {
+  /** User selection service */
+  const userSelectionService = inject(UserSelectionService).getSelections();
+
+  return inject(TissueOriginService).loadTissuePredictions(
+    userSelectionService.selectedOrganIri,
+    userSelectionService.selectedToolIri,
+  );
+}
+
+/**
+ * Injectable
+ */
+@Injectable({ providedIn: 'root' })
+export class TissueOriginService {
+  /** CSV file */
+  private file: File | null = null;
+
+  /** Sample CSV file */
+  private sampleFile?: File = undefined;
+
+  /** Prediction API endpoint */
+  private readonly endpoint = inject(TISSUE_ORIGIN_ENDPOINT);
+
+  /** http client to make http requests */
+  private readonly http = inject(HttpClient);
+
+  /** Updates current file with a new file  */
+  setFile(file: File): void {
+    this.file = file;
+  }
+
+  /** Sets current file to the Sample file */
+  async setSampleFile(): Promise<void> {
+    this.sampleFile ??= await this.loadSampleFile();
+    this.setFile(this.sampleFile);
+  }
+
+  /**
+   * Returns current file
+   * @returns File
+   */
+  getFile(): File | null {
+    return this.file;
+  }
+
+  /**
+   * Gets Predictions for the current csv file
+   * @returns Predictions array observable
+   */
+  loadTissuePredictions(selectedOrganIri?: string, selectedToolIri?: string): Observable<TissueOriginPredictions> {
+    if (this.file === null) {
+      return of({ sources: [], rui_locations: {} });
+    }
+
+    const body = {
+      ...(selectedOrganIri && { organ: selectedOrganIri }),
+      ...(selectedToolIri && { tool: selectedToolIri }),
+    };
+
+    return from(this.file.text()).pipe(
+      switchMap((data) =>
+        this.http.post<TissueOriginPredictions>(
+          `${this.endpoint}/cell-summary-report`,
+          {
+            ...body,
+            csvString: data,
+          },
+          {
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            responseType: 'json',
+          },
+        ),
+      ),
+    );
+  }
+
+  /**
+   * Loads CSV sample file
+   * @returns File promise
+   */
+  async loadSampleFile(): Promise<File> {
+    const content$ = this.http.get(SAMPLE_FILE_URL, {
+      responseType: 'text',
+    });
+    const content = await firstValueFrom(content$);
+    return new File([content], 'sample.csv', { type: 'text/csv' });
+  }
+
+  /**
+   * Gets supported tools
+   * @returns String array observable
+   */
+  loadSupportedTools(): Observable<SupportedTools[]> {
+    return this.http.get<SupportedTools[]>(`${this.endpoint}/supported-tools`);
```

Comment:
We need to update the requests to use the HraPopService from @hra-api/client instead of doing the http calls ourselves